doc/rpc/pathfinder_ws.json

```diff
@@ -0,0 +1,146 @@
+{
+    "openrpc": "1.2.6",
+    "info": {
+        "title": "Pathfinder websocket RPC API",
+        "version": "0.1",
+        "description": "Provides additional (pathfinder specific) methods over and above the Starknet RPC API"
+    },
+    "methods": [
+        {
+            "name": "pathfinder_subscribe",
+            "summary": "Open a new websocket subscription",
+            "description": "Creates a websocket stream which will fire events of the requested subscription type. Only available for websocket connections.",
+            "params": [
+                {
+                    "name": "kind",
+                    "summary": "The type of subscription",
+                    "required": true,
+                    "type": "string",
+                    "enum": [
+                        "newHeads"
+                    ]
+                }
+            ],
+            "result": {
+                "name": "subscription ID",
+                "description": "An identifier for this subscription stream used to associate events with this subscription.",
+                "type": "integer"
+            }
+        },
+        {
+            "name": "pathfinder_unsubscribe",
+            "summary": "Closes a websocket subscription",
+            "description": "Creates a websocket stream which will fire events of the requested subscription type. Only available for websocket connections.",
+            "params": [
+                {
+                    "name": "subscription ID",
+                    "summary": "The subscription to close",
+                    "required": true,
+                    "type": "integer"
+                }
+            ],
+            "result": {
+                "name": "Unsubscription result",
+                "description": "True if the unsubscription was successful",
+                "type": "bool"
+            }
+        },
+        {
+            "name": "pathfinder_subscription",
+            "summary": "A subscription event notification sent by the node.",
+            "result": {
+                "schema": {
+                    "subscription": {
+                        "name": "Subscription ID",
+                        "summary": "The subscription this event is for",
+                        "required": true,
+                        "type": "integer"
+                    },
+                    "result": {
+                        "summary": "The event data",
+                        "required": true,
+                        "oneOf": [
+                            {
+                                "summary": "Sent by 'newHeads' subscriptions",
+                                "$ref": "#/components/schemas/BLOCK_HEADER"
+                            }
+                        ]
+                    }
+                }
+            }
+        }
+    ],
+    "components": {
+        "contentDescriptors": {},
+        "schemas": {
+            "BLOCK_HEADER": {
```

Comment:
I was about to suggest to use an external `$ref` but I've looked at the various OpenRPC schema files we have and there is no `BLOCK_HEADER` with a 100% matching structure. This one is the only one referencing both eth & starknet gas prices. I also assume that's why the name of these two fields is the main difference.
I'm writing this comment just to double-check with you this is expected.

Reply:
The official RPC specification also undergoes many iterations, so being in better control over our own types.